### DISCUSSION TITLE
Difference to `cryptography` module?

### DISCUSSION THREAD

**Discussion by ValueRaider** [Legrandin/pycryptodome#693] [Q&A] [answered]
Why do both `cryptography` and `pycryptodome` exist?

Both have the same sales pitch - a self-contained package of cryptographic primitives. `cryptography` offers a high-level API, but that's no reason to duplicate effort on the low-level. Both have significant contributing communities (although `cryptography` is bigger) so there must be a good reason for both projects to exist separately. What am I missing?

Context: I'm adding some decryption functionality to a project and can't decide which module to use.

**Comment by crass** [comment, marked as answer]
One of the main differences is that `cryptography` uses [openssl as a backend](https://cryptography.io/en/latest/openssl/) whereas pycryptodome has no dependencies. According to [pycrypto's website](https://www.pycrypto.org/), `cryptography` is "recommended for new applications" and has a "newer API with fewer gotchas", while `pycryptodome` is "recommended for existing software that depends on PyCrypto" and is a "fork of PyCrypto. Most applications should run unmodified.". So the question becomes, is openssl an asset or a liability?
